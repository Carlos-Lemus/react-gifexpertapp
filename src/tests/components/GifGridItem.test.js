const { shallow } = require("enzyme");
const { GifGridItem } = require("../../components/GifGridItem");

describe("Probando el componente de <GifGridItem />", () => {

    const title = "title test";
    const url = "url test";
    
    const wrapper = shallow( <GifGridItem title={title} url={url} /> );

    test("Probando que se muestre correctamente el componente de <GifGridItem />", () => {

        expect( wrapper ).toMatchSnapshot();

    })

    test('p debe tener el valor de title que se le pasan por props', () => {
        
        const p = wrapper.find("p");

        expect( p.text() ).toBe(title);

    })
    
    test('el img debe tener los valores url y alt que se le pasan por props', () => {
        
        const img = wrapper.find("img");

        expect( img.prop("src") ).toBe( url );
        expect( img.prop("alt") ).toBe( title );

    })

    test('Debe tener la misma animacion que card animate__animated animate__fadeIn animate__delay-2', () => {
        
        const div = wrapper.find("div");

        expect( div.prop("className") ).toBe("card animate__animated animate__fadeIn animate__delay-2");
    })
    
    

});