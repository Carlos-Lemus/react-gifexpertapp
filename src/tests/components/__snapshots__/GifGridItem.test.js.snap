// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando el componente de <GifGridItem /> Probando que se muestre correctamente el componente de <GifGridItem /> 1`] = `
<div
  className="card animate__animated animate__fadeIn animate__delay-2"
>
  <img
    alt="title test"
    src="url test"
  />
  <p>
    title test
  </p>
</div>
`;
